Content.Server/Tools/ToolSystem.Welder.cs

```diff
@@ -224,6 +224,10 @@ private void OnWelderAfterInteract(EntityUid uid, WelderComponent welder, AfterI
                     SoundSystem.Play(welder.WelderRefill.GetSound(), Filter.Pvs(uid), uid);
                     target.PopupMessage(args.User, Loc.GetString("welder-component-after-interact-refueled-message"));
                 }
+                else if (welderSolution.AvailableVolume <= 0)
+                {
+                    target.PopupMessage(args.User, Loc.GetString("welder-component-already-full"));
+                }
```

Comment:
Shouldn't this be done above trans?

Also you should use popupsystem not the obsolete methods.

Reply:
Logically, it makes no difference. Doing it this way is better because `args.Handled` must be set to true, and so if I return early after checking `welderSolution.AvailableVolume <= 0` I would have to duplicate that code.

The code around this block uses the obsolete methods. Should I update those lines as well, even if they're unrelated to this change?

Reply:
you don't have to, but any new code should not use obsoleted methods

Reply:
Replaced use of deprecated popup function.